addon-test-support/create.js

```diff
@@ -184,16 +188,17 @@ export function create(definitionOrUrl, definitionOrOptions, optionsOrNothing) {
   let builder = {
     object: buildObject
   };
-
+  
   let page = Ceibo.create(definition, assign({ builder }, options));
 
   if (page) {
     page.render = render;
     page.setContext = setContext;
     page.removeContext = removeContext;
 
+    Ceibo.meta(page).pageObjectDefinition = definitionToStore;
```

Comment:
I don't know exactly how to do it yet, but... but I think we really need to do it in custom ceibo builders land in order recursively store definitions per instance

Reply:
wow.. seems like I have something working:
```js
function buildObject(node, blueprintKey, blueprint, defaultBuilder) {
  // here we check if we are going to create against the page object instance
  // in case if that's the case use its definition
  let definition;
  if (Ceibo.meta(blueprint)) {
    definition = Ceibo.meta(blueprint).__poDef__;
  } else {
    definition = blueprint;
  }

  blueprint = assign({}, dsl, definition);

  const [ instance, blueprintToApply ] = defaultBuilder(node, blueprintKey, blueprint, defaultBuilder);

  // persist definiion once we have an instance
  Ceibo.meta(instance).__poDef__ = definition;
  
  return [ instance, blueprintToApply ];
}
```
I've replicated the same trick to the `buildChainObject`.
I've just hacked it in node modules of my one tiny project and didn't test it seriously. But at the first glance seems like it works..

What do you think? can we use it instead create-only?

Reply:
Definitely the right track, which I've tailored into a working implementation. Above will fail here:

```javascript
let definition = {
  foo: {
      baz: "prop"
  }
};
let page = create(definition);
let pageComposer = create({
  somePage: page
});
```

because the initial blueprint isn't a page object, which means that `blueprintToApply` will contain the page object and will fire off getters in the next `processNode` when `buildObject` is called again and `blueprint = assign({}, blueprint);` will fire off a `getter` as soon as it encounters `somePage`.

That can be refactored away, but even still, there's a subtler problem. The first invocation of `buildObject` on 
```javascript
{
  somePage: page
}
```
will lead to an definition still containing page objects to be stored since the definition object isn't currently a page object. Any subsequent creation will would fail to compose this page object properly since the stored definition must be a pojo.

I find it much more straightforward to always sanitize the definitions as soon as possible such that we are dealing with pojos once in `Ceibo.create`. It's also more performant than adding those additional checks with then `buildObject` itself.